ext/http/01_http.js

```diff
@@ -341,7 +341,8 @@
   const _ws = Symbol("[[associated_ws]]");
 
   function upgradeWebSocket(request, options = {}) {
-    if (request.headers.get("upgrade") !== "websocket") {
+    const upgrade = request.headers.get("upgrade");
+    if (!upgrade || upgrade.toLowerCase() !== "websocket") {
```

Comment:
Please use "primordials" to do this check. You can import `StringPrototypeToLowerCase` (see top of file how), and then do `StringPrototypeToLowerCase(upgrade) !== "websocket"`.